js/script.js

```diff
@@ -0,0 +1,250 @@
+    image: '../images/4-image.png',
+    technologies: ['HTML', 'Bootstrap', 'Ruby'],
+    liveLink: 'https://akezeth.github.io/Portfolio/',
+    sourceLink: 'https://github.com/akezeth/Portfolio-setup-and-mobile-first',
+  },
+];
+
+// fuction to add element given tag name and class name
+const createElement = (tag, className) => {
+  const el = document.createElement(tag);
+  el.classList.add(className);
+  return el;
+};
+
+// function to creat ul and li elements given an array of items and class name of the ul element
+const createList = (items, className) => {
+  const ul = createElement('ul', className);
+  items.forEach((item) => {
+    const li = createElement('li', `${className}-item`);
+    li.innerText = item;
+    ul.appendChild(li);
+  });
+  return ul;
+};
+
+// fuction to create button element given the class name, id and text content of the button
+const createButton = (className, id, text) => {
+  const btn = createElement('button', className);
+  btn.id = id;
+  btn.innerText = text;
+  return btn;
+};
+
+const projectContainer = document.getElementById('projectContainer');
+
+const createtopProjectCon = (project1) => {
+  const topProjectEl = createElement('div', project1);
+
+  const project1Image = createElement('div', 'project1Image');
+  topProjectEl.appendChild(project1Image);
+
+  const image1 = createElement('img', 'image1');
+  image1.src = topProjectObj.image;
+  project1Image.appendChild(image1);
+
+  const project1Info = createElement('div', 'project1Info');
+  topProjectEl.appendChild(project1Info);
+
+  const project1Title = createElement('h3', 'project1Title');
+  project1Title.innerHTML = topProjectObj.title;
+  project1Info.appendChild(project1Title);
+
+  const project1Description = createElement('p', 'project1Description');
+  project1Description.innerText = topProjectObj.description;
+  project1Info.appendChild(project1Description);
+
+  const project1Languages = createList(topProjectObj.technologies, 'project1Languages');
+  project1Info.appendChild(project1Languages);
+
+  const project1LinkBtn = createButton('popupBtn', topProjectObj.id, 'See Project');
+  project1Info.appendChild(project1LinkBtn);
+
+  return topProjectEl;
+};
+
+const createOtherProject = (otherProject) => {
+  const otherProjectEl = createElement('div', 'otherProject');
+
+  const otherProjectInfo = createElement('div', 'otherProjectInfo');
+  otherProjectEl.appendChild(otherProjectInfo);
+
+  const otherProjectTitle = createElement('h3', 'otherProjectTitle');
+  otherProjectTitle.innerHTML = otherProject.title;
+  otherProjectInfo.appendChild(otherProjectTitle);
+
+  const otherProjectDescription = createElement('p', 'otherProjectDescription');
+  otherProjectDescription.innerText = otherProject.description;
+  otherProjectInfo.appendChild(otherProjectDescription);
+
+  // const otherProjectLanguages = createList(projectArr.technologies, 'otherProjectLanguages');
+  // otherProjectInfo.appendChild(otherProjectLanguages);
+
```

Comment:
- [ ] Please consider removing `commented codes` as it will make your codebase messy, Remember if you ever like to see the codes you have written once you can use your commit history.